File: problema.rb
# Números perfectos
def es_numero_perfecto(n)
    (1...n).select { |i| n % i == 0 }.sum == n
  end
  
  def encontrar_numeros_perfectos(limite)
    (2...limite).each do |num|
      puts num if es_numero_perfecto(num)
    end
  end
  
  if __FILE__ == $0
    encontrar_numeros_perfectos(10_000)
  end
  

File: problema.py
#numeros perfecos
def es_numero_perfecto(n):
    return sum(i for i in range(1, n) if n % i == 0) == n

def encontrar_numeros_perfectos(limite):
    for num in range(2, limite):
        if es_numero_perfecto(num):
            print(num)

if __name__ == "__main__":
    encontrar_numeros_perfectos(10000)
